numpy/ma/tests/test_core.py

```diff
@@ -4262,6 +4262,29 @@ def test_default_fill_value_complex():
     # regression test for Python 3, where 'unicode' was not defined
     assert_(default_fill_value(1 + 1j) == 1.e20 + 0.0j)
 
+
+def test_sum_keepdims():
+
+    # Regression test for numpy/numpy#7312 which was due to masked_array.sum
+    # not supporting the keepdims option.
+
+    # First make sure keepdims works when called explicitly
+    result = np.sum(np.ma.masked_array([1,2,3],mask=[0,0,1]), keepdims=True)
```

Comment:
Lots of PEP8 violations in this function (spaces after commas) -- could you fix that? Otherwise LGTM